### PR TITLE
ref(abuse) Add the parent_api back to the metrics

### PR DESCRIPTION
**Pull request by evanh**
The parent_api metric can be useful to see which feature has spiked in abuse,
since one feature can call many referrers and make it harder to see changes in
traffic usage.

This metric was removed in the past because we found cases of the endpoint not
being correct. Referrers were attached to endpoints that didn't actually call
the referrer. However the Sentry SDK has improved quite a bit since then and
even if the endpoint is not correct, it can still be useful signal of a problem.

### CODE REVIEW

File: tests/subscriptions/entity_subscriptions/test_entity_subscriptions_data.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def test_entity_subscriptions_data() -> None:
         RedisSubscriptionDataStore(redis_client, entity_key, PartitionId(i))
         for i in range(topic_spec.partitions_number)
     ]
-    print(stores[0].all())
```
Comment (evanh) [inline tests/subscriptions/entity_subscriptions/test_entity_subscriptions_data.py:55]: This is unrelated, I just happened to notice this extra print.

File: snuba/query/query_settings.py [hunk 2]
```diff
@@ -91,6 +101,9 @@ def get_consistent(self) -> bool:
     def get_debug(self) -> bool:
         return self.__debug
 
+    def get_parent_api(self) -> str:
```
Comment (volokluev) [inline snuba/query/query_settings.py:104]: why does this need to be in the query settings?

**Review by volokluev — COMMENTED** [review]
So once this merges, will it be automatically populated?The sentry side has been sending it all this time and we have just been ignoring it?

File: snuba/query/query_settings.py (hunk 2 above)
Comment (evanh) [reply snuba/query/query_settings.py]: I needed it in order to get some tests passing, but I think I was in a "Friday afternoon" haze and I did something wrong. I'm changing it.

File: snuba/request/schema.py [hunk 3]
```diff
@@ -175,7 +175,7 @@ def generate_template(self) -> Any:
             # TODO: move this to attribution
             "legacy": {"type": "boolean", "default": False},
             "referrer": {"type": "string", "default": "<unknown>"},
-            "parent_api": {"type": "string", "default": "<unknown>"},
+            # "parent_api": {"type": "string", "default": "<unknown>"},
```
Comment (volokluev) [inline snuba/request/schema.py:178]: should this be commented out?

**Comment by evanh** [comment]
> So once this merges, will it be automatically populated?The sentry side has been sending it all this time and we have just been ignoring it?

No, there is an SDK change that requires a new version, which requires a pypi change, and then Sentry needs to be updated.

**Review by lynnagara — APPROVED** [review]
